projects/planner/src/app/shared/blend-pricing.service.spec.ts

```diff
@@ -0,0 +1,324 @@
+        };
+
+        const result: BlendPrice = {
+            o2Price: sut.o2Price,
+            hePrice: sut.hePrice,
+            topMixPrice: sut.topMixPrice,
+            totalPrice: sut.totalPrice
+        };
+
+        expect(result).toEqual(expected);
+    });
+
+    it('should calculate prices when units prices are positive numbers and amounts of gases are negative numbers', () => {
+        sut.o2UnitPrice = 1;
+        sut.heUnitPrice = 2;
+        sut.topMixUnitPrice = 3;
+
+        gasBlender.addO2 = -5;
+        gasBlender.addHe = -6;
+        gasBlender.addTop = -7;
+
+        sut.calculate();
+
+        const expected: BlendPrice = {
+            o2Price: -5,
+            hePrice: -12,
+            topMixPrice: -21,
+            totalPrice: -38
+        };
+
+        const result: BlendPrice = {
+            o2Price: sut.o2Price,
+            hePrice: sut.hePrice,
+            topMixPrice: sut.topMixPrice,
+            totalPrice: sut.totalPrice
+        };
+
+        expect(result).toEqual(expected);
+    });
+
+    it('should calculate totalPrice when one unit price is positive, other are negative and amounts of gases are negative numbers', () => {
+        sut.o2UnitPrice = -1;
+        sut.heUnitPrice = 2;
+        sut.topMixUnitPrice = -3;
+
+        gasBlender.addO2 = -4;
+        gasBlender.addHe = -5;
+        gasBlender.addTop = -6;
+
+        sut.calculate();
+
+        const expected: BlendPrice = {
+            o2Price: 4,
+            hePrice: -10,
+            topMixPrice: 18,
+            totalPrice: 12
+        };
+
+        const result: BlendPrice = {
+            o2Price: sut.o2Price,
+            hePrice: sut.hePrice,
+            topMixPrice: sut.topMixPrice,
+            totalPrice: sut.totalPrice
+        };
+
+        expect(result).toEqual(expected);
+    });
+
+    it('should calculate prices when all unit prices and gas amounts are negative', () => {
+        sut.o2UnitPrice = -1;
+        sut.heUnitPrice = -2;
+        sut.topMixUnitPrice = -3;
+
+        gasBlender.addO2 = -4;
+        gasBlender.addHe = -5;
+        gasBlender.addTop = -6;
+
+        sut.calculate();
+
+        const expected: BlendPrice = {
+            o2Price: 4,
+            hePrice: 10,
+            topMixPrice: 18,
+            totalPrice: 32
+        };
+
+        const result: BlendPrice = {
+            o2Price: sut.o2Price,
+            hePrice: sut.hePrice,
+            topMixPrice: sut.topMixPrice,
+            totalPrice: sut.totalPrice
+        };
+
+        expect(result).toEqual(expected);
+    });
+
+    it('should calculate prices when unit prices and gas amounts are very small values', () => {
+        sut.o2UnitPrice = 0.000001;
+        sut.heUnitPrice = 0.000002;
+        sut.topMixUnitPrice = 0.000003;
+
+        gasBlender.addO2 = 0.000004;
+        gasBlender.addHe = 0.000005;
+        gasBlender.addTop = 0.000006;
+
+        sut.calculate();
+
+        const expected: BlendPrice = {
```

Comment:
Javascript is unable to count with precise numbers, use less precission instead. 6 decimals is enouth.
I.e. remove this test.